File: 030_Palindrome_Remove_Occurrences/Remove_Occurrences.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string removeOccurrences(string s, string part) {
        while (s.length() > 0 && s.find(part) < s.length()) {
            s.erase(s.find(part), part.length());
        }

        // Better and cleaner condition:
        // while (s.find(part) != string::npos) {
        //     s.erase(s.find(part), part.length());
        // }

        return s;
    }
};

int main() {
    Solution obj;
    string s = "daabcbaabcbc";
    string part = "abc";

    cout << obj.removeOccurrences(s, part);
    return 0;
}

File: 030_Palindrome_Remove_Occurrences/Remove_Occurrences.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        while part in s:
            s = s.replace(part, "", 1)  # remove one occurrence at a time
        return s


# Example usage
obj = Solution()
s = "daabcbaabcbc"
part = "abc"
print(obj.removeOccurrences(s, part))
